features/breed_detail/src/test/java/com/example/catbreeds/breed_detail/unitTests/BreedDetailViewModelTest.kt

```diff
@@ -0,0 +1,169 @@
+package com.example.catbreeds.breed_detail.unitTests
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.lifecycle.SavedStateHandle
+import com.example.catbreeds.domain.models.Breed
+import com.example.catbreeds.domain.repository.BreedRepository
+import com.example.catbreeds.breed_detail.BreedDetailViewModel
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.every
+import io.mockk.mockk
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.StandardTestDispatcher
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.junit.After
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+// Mock data
+object TestBreedDetailData {
+    fun getTestBreed(id: String, isFav: Boolean = false) = Breed(
+        id = id,
+        name = "Siberian $id",
+        origin = "Russia",
+        temperament = "Active, Playful",
+        life_span = "10 - 15",
+        description = "A strong, moderately large cat.",
+        reference_image_id = "sibe",
+        isFavorite = isFav
+    )
+    val testBreed = getTestBreed("testId", false)
+}
+
+@ExperimentalCoroutinesApi
+class BreedDetailViewModelTest {
+
+    @get:Rule
+    val instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    private val testDispatcher = StandardTestDispatcher()
+    private lateinit var breedRepository: BreedRepository
+
+    @Before
+    fun setup() {
+        Dispatchers.setMain(testDispatcher)
+        breedRepository = mockk(relaxed = true)
+    }
+
+    @After
+    fun tearDown() {
+        Dispatchers.resetMain()
+    }
+
+    @Test
+    fun `WHEN initialized with valid ID SHOULD load breed`() = runTest {
+        // GIVEN
+        val targetBreed = TestBreedDetailData.testBreed
+        val breedId = targetBreed.id
+        val savedStateHandle = SavedStateHandle(mapOf("breedId" to breedId))
+        coEvery { breedRepository.getBreedById(breedId) } returns targetBreed
+        every { breedRepository.getFavoriteBreeds() } returns flowOf(emptyList())
+
+        // WHEN
+        val viewModel = BreedDetailViewModel(breedRepository, savedStateHandle)
+        testDispatcher.scheduler.advanceUntilIdle()
```

Comment:
just calling `advanceUntilIdle()` is enough as the `runTest` is already running on the test dispatcher.
Same for all other calls.
```suggestion
        advanceUntilIdle()
```